homeassistant/components/geniushub/climate.py

```diff
@@ -0,0 +1,204 @@
+        GeniusClimate._genius_hub = genius_hub
+        self._name = zone['name']
+        self._device_id = zone['iID']
+        self._current_temperature = zone['current_temperature']
+        self._target_temperature = zone['target_temperature']
+        self._mode = zone['mode']
+        self._is_active = zone['is_active']
+
+    @property
+    def state(self):
+        """Return the current state."""
+        if self._is_active:
+            return STATE_ON
+
+        return STATE_OFF
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._name
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def min_temp(self):
+        """Return max valid temperature that can be set."""
+        return 4.0
+
+    @property
+    def max_temp(self):
+        """Return max valid temperature that can be set."""
+        return 28.0
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return OPERATION_LIST
+
+    @property
+    def is_on(self):
+        """Return true if the device is on."""
+        if self._mode == "off":
+            return False
+
+        return True
+
+    @property
+    def is_away_mode_on(self):
+        """Return true if away mode is on."""
+        if self._mode == "away":
+            return True
+
+        return False
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        return self.get_current_operation_mode(self._mode)
+
+    @staticmethod
+    def get_current_operation_mode(mode):
+        """Return the current operational mode."""
+        mode_map = {
+            'override': STATE_HEAT,
+            'footprint': STATE_ECO,
+            'timer': STATE_AUTO,
+        }
+        return mode_map.get(mode, STATE_IDLE)
+
+    def get_operation_mode(self, operation_mode):
+        """Coverts operation mode from Home Assistant to Genius Hub."""
+        # These needed to be mapped into HA modes:
+        # Off       => OFF      => STATE_IDLE   # Mode_Off: 1,
+        # Override  => HEAT     => STATE_HEAT # Mode_Boost: 16,
+        # Footprint => ECO      => STATE_ECO    # Mode_Footprint: 4,
+        # Timer     => AUTO     => STATE_AUTO   # Mode_Timer: 2,
+        # Away                      # Mode_Away: 8,
+        #
+        # OPERATION_LIST = [STATE_IDLE, STATE_HEAT, STATE_ECO, STATE_AUTO]
+        operation_mode_map = {
```

Comment:
See above.

Reply:
What about fBootSP: self._target_temperature?

Reply:
Right, I missed that.